### PR TITLE
f-contact-preferences@v0.9.0 - Register Module and Initialise Page based on Authorisation

### PR DESCRIPTION
**Pull request by oliversweb**
### Added
- Expose new isAuthFinished `required` property

### Changed
- Registered the Store Module on Create()
- Initialise Page based on isAuthFinished

### CODE REVIEW

File: packages/components/pages/f-contact-preferences/src/components/ContactPreferences.vue [hunk 1]
```diff
@@ -142,8 +147,25 @@ export default {
         ])
     },
 
+    watch: {
+        immediate: true,
```
Comment (SachaZvetelman) [inline packages/components/pages/f-contact-preferences/src/components/ContactPreferences.vue]: Does this work correctly? I thought `immediate` had to be inside the bit you're watching, and not outside.

File: packages/components/pages/f-contact-preferences/src/components/_tests/ContactPreferences.test.js [hunk 2]
```diff
@@ -143,6 +166,34 @@ describe('ContactPreferences Component', () => {
...
+
+        it('should not call initialise method if the authorisation has not completed', () => {
+            // Arrange
+            sutProps = { ...sutProps, isAuthFinished: false };
+
+            // Act
+            wrapper = mountContactPreferences();
+
+            // Assert
+            expect(storeActions.loadPreferences).not.toHaveBeenCalled();
```
Comment (SachaZvetelman) [inline packages/components/pages/f-contact-preferences/src/components/_tests/ContactPreferences.test.js]: The test's name refers to the initialise method, but the assert refers to something else. It's a bit confusing 😅

File: packages/components/pages/f-contact-preferences/src/components/_tests/ContactPreferences.test.js [hunk 3]
```diff
@@ -143,6 +166,34 @@ describe('ContactPreferences Component', () => {
...
+        });
+
+        it('should only call initialise method once the authorisation has completed', async () => {
```
Comment (SachaZvetelman) [inline packages/components/pages/f-contact-preferences/src/components/_tests/ContactPreferences.test.js:189]: If I understand correctly, I think in this test what you're trying to check is that when your watcher sees that the authorisation is complete, the initialise method is called.

To me, this test would look something like this (kinda pseudo-code):

```
// Arrange
wrapper = mountContactPreferences();

// Act
wrapper.vm.$options.watch.isAuthFinished.call(wrapper.vm, true);

// Assert
expect(initialiseSpy).toHaveBeenCalled();
```
More info: https://github.com/vuejs/vue-test-utils/issues/331#issuecomment-382037200

File: packages/components/pages/f-contact-preferences/stories/ContactPreferences.stories.js [hunk 4]
```diff
@@ -72,11 +68,12 @@ export const ContactPreferencesComponent = () => ({
...
     template: '<contact-preferences ' +
     ':authToken="authToken" ' +
+    ':isAuthFinished="true" ' +
     ':locale="locale" ' +
-    ':smart-gateway-base-url="smartGatewayBaseUrl" ' +
+    'smart-gateway-base-url="" ' +
     ':prepareStory="prepareStory" ' +
     // eslint-disable-next-line no-template-curly-in-string
-    ' :key="`${authToken},${locale},${smartGatewayBaseUrl}`" />'
+    ' :key="`${authToken},${locale}`" />'
```
Comment (SachaZvetelman) [inline packages/components/pages/f-contact-preferences/stories/ContactPreferences.stories.js]:
```suggestion
    ':key="`${authToken},${locale}`" />'
```

File: packages/components/pages/f-contact-preferences/src/components/_tests/ContactPreferences.test.js (hunk 3 above)
Comment (oliversweb) [reply packages/components/pages/f-contact-preferences/src/components/_tests/ContactPreferences.test.js:189]: @SachaZvetelman I want to mount the component with the `isAuthFinished` = false then check that the `initialise()` has not been called then set the prop and recheck that the `initialise()` has now been called. If I don't check it wasn't called on the `mount` I cannot prove the check; after setting the prop to `true`, was not for the previous `mount` action.

File: packages/components/pages/f-contact-preferences/src/components/ContactPreferences.vue (hunk 1 above)
Comment (oliversweb) [reply packages/components/pages/f-contact-preferences/src/components/ContactPreferences.vue]: There is an alternative way ... I will try that 👍

File: packages/components/pages/f-contact-preferences/src/components/_tests/ContactPreferences.test.js (hunk 2 above)
Comment (oliversweb) [reply packages/components/pages/f-contact-preferences/src/components/_tests/ContactPreferences.test.js]: It was simpler to just check if a method inside the `Initialise()` had been called but I can set up a new spy 👍

File: packages/components/pages/f-contact-preferences/stories/ContactPreferences.stories.js (hunk 4 above)
Comment (oliversweb) [reply packages/components/pages/f-contact-preferences/stories/ContactPreferences.stories.js]: I think it was always like that do didn't want to break anything but can remove 👍

File: packages/components/pages/f-contact-preferences/src/components/_tests/ContactPreferences.test.js (hunk 3 above)
Comment (SachaZvetelman) [reply packages/components/pages/f-contact-preferences/src/components/_tests/ContactPreferences.test.js:189]: If you comment out the watcher, does this test still pass? I'm not sure. If it fails when you do that, I'm ok to approve. Could you please check that?

Comment (oliversweb) [reply packages/components/pages/f-contact-preferences/src/components/_tests/ContactPreferences.test.js:189]: @SachaZvetelman I comment out the watcher and it failed

**Review by SachaZvetelman — APPROVED** [review]

**Review by JoshuaNg2332 — APPROVED** [review]

**Pull request merged**